vortex-array/src/stream/ext.rs

```diff
@@ -21,7 +21,7 @@ pub trait ArrayStreamExt: ArrayStream {
         }
     }
 
-    fn take_rows(self, indices: &Array) -> VortexResult<impl ArrayStream>
+    fn take_rows(self, indices: Array) -> VortexResult<impl ArrayStream>
```

Comment:
taking indices by reference doesn't work if you want to return the stream instead of immediately evaluating it